### PR TITLE
docs: Improve documentation on loops

### PR DESCRIPTION
**Pull request by Joibel**
The loops documentation is 'by example' only. I have attempted to improve this to explain what is going on, and why someone might use each form of loops

This is a documentation only change. The examples are unchanged.

### CODE REVIEW

File: docs/walk-through/loops.md [hunk 1]
```diff
@@ -69,7 +79,7 @@ spec:
       args: [/etc/os-release]
 ```
 
-We can pass lists of items as parameters:
+This example does exactly the same job as the previous example, but using `withParam` to pass the information as a JSON array argument, instead of directly hardcoded into the template.
```
Comment (tico24) [inline docs/walk-through/loops.md:82]: perhaps "...instead of being directly hardcoded..."? or "hard-coding"?

**Review by tico24 — APPROVED** [review]
One small nit from me.

File: docs/walk-through/loops.md (hunk 1 above)
Comment (Joibel) [reply docs/walk-through/loops.md]: Done.

**Review by tico24 — APPROVED** [review]

**Review by alexec — APPROVED** [review]
LGTM

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**